trochilidae/tr_network.c

```
@@ -65,15 +65,17 @@ struct in_addr find_ip_address(const char *domain) {
  * @return
  */
 extern DomainPortEntry * parse_domain_port_pairs(const char* input, int* numPairs) {
-    DomainPortEntry* pairs = (DomainPortEntry*)malloc(strlen(input) * sizeof(DomainPortEntry));
+    char *input_dup = strdup(input);
+    DomainPortEntry* pairs = (DomainPortEntry*)malloc(strlen(input_dup) * sizeof(DomainPortEntry));
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)